docs/authoring.md

```diff
@@ -23,45 +25,69 @@ Accepted `<TargetFramework>` properties |
-We are working on implementing diagnostics in our tool that will catch these errors before a winmd is generated for your component.
-
-2. Not all C# types have been mapped to WinRT types, but we are working on completing this coverage. 
-
-3. Composable type support is still in progress
+## Using your authored component
+To use the component in a C# app, the authored component just needs to be added as a project/package reference.
 
-## Using the authored component
+For native (C++) apps, there are DLLs needed to host your authored component. When you package your runtime component, they are automatically
+added to your nupkg, before the ```GenerateNuspec``` MSBuild step.  
 
-To use the component in a C# app, the authored component just needs to be added as a project/package reference.
+You will need to create a targets file for your component, if you are not already, that imports a CsWinRT targets file. The imported targets file configures the native app to use the hosting dlls at runtime.
+This means for your component ```MyAuthoredComponent```, you will need a targets file that has an import statment for ```MyAuthoredComponent.CsWinRT.targets```. 
+The targets file you need **must** be named ```MyAuthoredComponent.targets```, otherwise NuGet will ignore it.
 
-For native (C++) apps, more steps are needed.
+For example, the simplest definition of ```MyAuthoredComponent.targets``` would be:
+``` targets
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+	<Import Project="$(MSBuildThisDirectory)MyAuthoredComponent.CsWinRT.targets"	/>
+</Project>
+```
 
-Modifications needed: 
-  1. Reference to your authored component in the consuming app
-  2. Add a manifest file and runtimeconfig file [for native consumption only] 
-  3. Use a MSBuild targets for copying necessary DLLs
+The ```MyAuthoredComponent.CsWinRT.targets``` is added to the package by CsWinRT, you'll just need to add your ```MyAuthoredComponent.targets``` file to the package as well.
+Do this by adding the following to ``MyAuthoredComponent.csproj```
 
+``` csproj
+<ItemGroup>
+  <_PackageFiles Include="MyAuthoredComponent.targets" PackagePath="build;buildTransitive"/>
+</ItemGroup>
+```
 
-### Add the reference to your authored component
-**In Visual Studio** Under the Project node, right click on "References", click "Add Reference", then "Browse" and add the `.winmd` file generated in your authored component. 
+**If you are going to write your own nuspec** then the CsWinRT target that adds the hosting dlls to your package will not run, and you should make sure your nuspec contains the following ```file``` entries for ```MyAuthoredComponent``` (note: your TargetFramework may vary).
+
+``` nuspec
+  <files>
+    <file src="build\MyAuthoredComponent.CsWinRT.targets"             target="build\MyAuthoredComponent.CsWinRT.targets" />

```

Comment:
Are the src paths here correct?  I feel like that is not the directory structure of the files.

Reply:
Updated